packages/legend-query/src/components/QueryBuilderPropertySearchPanel.tsx

```diff
@@ -0,0 +1,717 @@
+    const propertySearchPanelState = explorerState.propertySearchPanelState;
+    const [{ isDragging }, dragConnector] = useDrag(
+      () => ({
+        type:
+          node instanceof QueryBuilderExplorerTreePropertyNodeData
+            ? node.type instanceof Enumeration
+              ? QUERY_BUILDER_EXPLORER_TREE_DND_TYPE.ENUM_PROPERTY
+              : node.type instanceof Class
+              ? QUERY_BUILDER_EXPLORER_TREE_DND_TYPE.CLASS_PROPERTY
+              : QUERY_BUILDER_EXPLORER_TREE_DND_TYPE.PRIMITIVE_PROPERTY
+            : QUERY_BUILDER_EXPLORER_TREE_DND_TYPE.ROOT,
+        item: (): { node?: QueryBuilderExplorerTreePropertyNodeData } =>
+          node instanceof QueryBuilderExplorerTreePropertyNodeData
+            ? { node }
+            : {},
+        collect: (monitor): { isDragging: boolean } => ({
+          isDragging: monitor.isDragging(),
+        }),
+      }),
+      [node],
+    );
+    const isMultiple =
+      (node instanceof QueryBuilderExplorerTreePropertyNodeData &&
+        (node.property.multiplicity.upperBound === undefined ||
+          node.property.multiplicity.upperBound > 1)) ||
+      (node instanceof QueryBuilderExplorerTreeSubTypeNodeData &&
+        (node.multiplicity.upperBound === undefined ||
+          node.multiplicity.upperBound > 1));
+    queryBuilderState.explorerState.propertySearchPanelState.setIsSearchPanelOpen(
+      !isDragging,
+    );
+    const treeData = explorerState.nonNullableTreeData;
+    const onNodeSelect = (
+      selectedNode: QueryBuilderExplorerTreeNodeData,
+    ): void => {
+      if (selectedNode.childrenIds.length) {
+        selectedNode.isOpen = true;
+        if (
+          (selectedNode instanceof QueryBuilderExplorerTreePropertyNodeData ||
+            selectedNode instanceof QueryBuilderExplorerTreeSubTypeNodeData) &&
+          selectedNode.type instanceof Class
+        ) {
+          (selectedNode instanceof QueryBuilderExplorerTreeSubTypeNodeData
+            ? getAllOwnClassProperties(selectedNode.type)
+            : getAllClassProperties(selectedNode.type).concat(
+                getAllClassDerivedProperties(selectedNode.type),
+              )
+          ).forEach((property) => {
+            const propertyTreeNodeData = getQueryBuilderPropertyNodeData(
+              queryBuilderState.graphManagerState,
+              property,
+              selectedNode,
+              guaranteeNonNullable(queryBuilderState.querySetupState.mapping),
+            );
+            treeData.nodes.set(propertyTreeNodeData.id, propertyTreeNodeData);
+          });
+          selectedNode.type._subclasses.forEach((subclass) => {
+            const subTypeTreeNodeData = getQueryBuilderSubTypeNodeData(
+              subclass,
+              node,
+            );
+            treeData.nodes.set(subTypeTreeNodeData.id, subTypeTreeNodeData);
+          });
+        }
+      }
+      explorerState.refreshTree();
+    };
+    const showSelectedNodeInTree = (): void => {
```

Comment:
should we close the search panel once the user selects to see the node in the tree?

Reply:
we do close it once user selects to see the node in the tree